### PR TITLE
fix(CBH): fix cbh instance

### PR DESCRIPTION
**Pull request by houpeng80**
**What this PR does / why we need it**:
  fix cbh instance and data source
**Which issue this PR fixes**:
*(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*
fixes #xxx

**Special notes for your reviewer**:

**Release note**:

```release-note
  fix cbh instance and data source
```

## PR Checklist

* [ ] Tests added/passed.
* [ ] Documentation updated.
* [ ] Schema updated.

## Acceptance Steps Performed

```
make testacc TEST='./huaweicloud/services/acceptance/cbh/'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/services/acceptance/cbh/ -v  -timeout 360m -parallel 4
=== RUN   TestAccDatasourceCbhInstances_basic 
=== PAUSE TestAccDatasourceCbhInstances_basic 
=== RUN   TestAccCBHInstance_basic            
=== PAUSE TestAccCBHInstance_basic            
=== CONT  TestAccDatasourceCbhInstances_basic 
=== CONT  TestAccCBHInstance_basic            
--- PASS: TestAccDatasourceCbhInstances_basic (1094.94s) 
--- PASS: TestAccCBHInstance_basic (1138.91s) 
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/cbh       1138.961s
```

### CODE REVIEW

File: docs/resources/cbh_instance.md [hunk 1]
```diff
@@ -62,6 +62,11 @@ The following arguments are supported:
 
   Changing this parameter will create a new resource.
 
+* `product_info` - (Required, List, ForceNew)
```
Comment (deer-hang) [inline docs/resources/cbh_instance.md]: How about add a description to this field

Comment (houpeng80) [reply docs/resources/cbh_instance.md]: done

File: huaweicloud/services/cbh/data_source_huaweicloud_cbh_instance.go [hunk 2]
```diff
@@ -192,11 +193,6 @@ func CbhInstancesInstanceSchema() *schema.Resource {
 				Computed:    true,
 				Description: `Indicates the type of the bastion.`,
 			},
-			"auto_renew": {
```
Comment (ShiChangkuo) [inline huaweicloud/services/cbh/data_source_huaweicloud_cbh_instance.go]: the docs should be updated

File: docs/resources/cbh_instance.md [hunk 3]
```diff
@@ -62,6 +62,11 @@ The following arguments are supported:
 
   Changing this parameter will create a new resource.
 
+* `product_info` - (Required, List, ForceNew) Specifies the product info of the instance.
```
Comment (ShiChangkuo) [inline docs/resources/cbh_instance.md]: the `product_info` has marked be required, but it does not used in example above

**Comment by houpeng80** [comment]
make testacc TEST='./huaweicloud/services/acceptance/cbh/' TESTARGS='-run TestAccCBHInstance_basic'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/services/acceptance/cbh/ -v -run TestAccCBHInstance_basic -timeout 360m -parallel 4
=== RUN   TestAccCBHInstance_basic
=== PAUSE TestAccCBHInstance_basic
=== CONT  TestAccCBHInstance_basic
--- PASS: TestAccCBHInstance_basic (1098.57s) 
PASS                                                                                                              
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/cbh       1098.618s

File: docs/resources/cbh_instance.md [hunk 4]
```diff
@@ -9,21 +9,24 @@ Manages CBH instance resources within HuaweiCloud.
...
-  name              = "cbh_instance_test"
+  flavor_id         = "cbh.basic.10"
+  name              = var.name
   vpc_id            = var.vpc_id
-  availability_zone = var.availability_zone
-  hx_password       = "test_123456",
-  bastion_type      = "OEM"
   subnet_id         = var.subnet_id
   security_group_id = var.security_group_id
+  availability_zone = "cn-north-4a"
```
Comment (Lance52259) [inline docs/resources/cbh_instance.md]: do not use the specified az.
using
```
availability_zone = var.availability_zone
```
or
```
availability_zone = data.huaweicloud_availability_zone.test.names[0]
```

File: docs/resources/cbh_instance.md [hunk 5]
```diff
@@ -80,74 +79,31 @@ The following arguments are supported:
...
-
-* `user_data` - (Optional, String, ForceNew) Specifies the inject user data.
-
-  Changing this parameter will create a new resource.
-
-* `password` - (Optional, String) Specifies the initial password.
+* `subnet_id` - (Required, String) Specifies the ID of a subnet.
 
-* `key_name` - (Optional, String, ForceNew) Specifies the secret key of the admin.
+* `security_group_id` - (Required, String) Specifies the ID list of the security group.
```
Comment (Lance52259) [inline docs/resources/cbh_instance.md]: ID list? not ID?

**Comment by houpeng80** [comment]
make testacc TEST='./huaweicloud/services/acceptance/cbh/' TESTARGS='-run TestAccCBHInstance_basic'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/services/acceptance/cbh/ -v -run TestAccCBHInstance_basic -timeout 360m -parallel 4
=== RUN   TestAccCBHInstance_basic
=== PAUSE TestAccCBHInstance_basic
=== CONT  TestAccCBHInstance_basic
--- PASS: TestAccCBHInstance_basic (1100.74s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/cbh       1100.788s

**Comment by houpeng80** [comment]
make testacc TEST='./huaweicloud/services/acceptance/cbh/' TESTARGS='-run TestAccCBHInstance_basic'
==> Checking that code complies with gofmt requirements...
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/services/acceptance/cbh/ -v -run TestAccCBHInstance_basic -timeout 360m -parallel 4
=== RUN   TestAccCBHInstance_basic
=== PAUSE TestAccCBHInstance_basic
=== CONT  TestAccCBHInstance_basic
--- PASS: TestAccCBHInstance_basic (1366.49s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/cbh       1366.538s

File: docs/resources/cbh_instance.md [hunk 6]
```diff
@@ -4,26 +4,29 @@ subcategory: "Cloud Bastion Host (CBH)"
 
 # huaweicloud_cbh_instance
 
-Manages CBH instance resources within HuaweiCloud.
+Manages a CBH instance resources within HuaweiCloud.
```
Comment (deer-hang) [inline docs/resources/cbh_instance.md]: resources/resource

File: docs/resources/cbh_instance.md [hunk 7]
```diff
@@ -4,26 +4,29 @@ subcategory: "Cloud Bastion Host (CBH)"
...
-  name              = "cbh_instance_test"
+  flavor_id         = "cbh.basic.10"
+  name              = var.name
   vpc_id            = var.vpc_id
-  availability_zone = var.availability_zone
-  hx_password       = "test_123456",
-  bastion_type      = "OEM"
   subnet_id         = var.subnet_id
   security_group_id = var.security_group_id
+  availability_zone = var.availability_zone
```
Comment (deer-hang) [inline docs/resources/cbh_instance.md:25]: make the variable value together

File: docs/resources/cbh_instance.md [hunk 8]
```diff
@@ -69,203 +68,54 @@ The following arguments are supported:
...
 
   Changing this parameter will create a new resource.
 
-* `ip_address` - (Optional, String) Specifies the IP address of the subnet.
+* `auto_renew` - (Optional, String) Specifies whether auto renew is enabled.
+  Valid values are **true** and **false**. Defaults to **false**.
 
-* `public_ip` - (Optional, List)
-  The [PublicIP](#CBHInstance_PublicIP) structure is documented below.
+* `subnet_address` - (Optional, String) Specifies the IP address of the subnet.
```
Comment (deer-hang) [inline docs/resources/cbh_instance.md:86]: Add "If not specified, a new IP address will be assigned"

File: docs/resources/cbh_instance.md [hunk 9]
```diff
@@ -69,203 +68,54 @@ The following arguments are supported:
...
 
-* `root_volume` - (Optional, List, ForceNew)
+* `password` - (Required, String) Specifies the password for logging in to the management console.
```
Comment (deer-hang) [inline docs/resources/cbh_instance.md]: make the required field before the Optional

File: docs/resources/cbh_instance.md [hunk 10]
```diff
@@ -69,203 +68,54 @@ The following arguments are supported:
...
-
-  Changing this parameter will create a new resource.
-
-* `resource_spec_code` - (Optional, String, ForceNew) Specifies the resource specification.
+* `public_ip_id` - (Optional, String) Specifies the ID of the elastic IP.
 
-  Changing this parameter will create a new resource.
+* `public_ip` - (Optional, String) Specifies the elastic IP address.
 
 * `ipv6_enable` - (Optional, Bool, ForceNew) Specifies whether the IPv6 network is enabled.
```
Comment (deer-hang) [inline docs/resources/cbh_instance.md]: Add `Default to false`

File: huaweicloud/services/cbh/data_source_huaweicloud_cbh_instance.go [hunk 11]
```diff
@@ -237,7 +238,7 @@ func resourceCbhInstancesRead(ctx context.Context, d *schema.ResourceData, meta
 	if err != nil {
 		return diag.FromErr(err)
 	}
-	instances := flattenGetInstancesResponseBodyInstance(getCbhInstancesRespBody)
+	instances := utils.PathSearch("instance", getCbhInstancesRespBody, make([]interface{}, 0)).([]interface{})
```
Comment (deer-hang) [inline huaweicloud/services/cbh/data_source_huaweicloud_cbh_instance.go:241]: The list response value is unequal with schema, such as `instanceId`, `publicip`, `vpcID` and `subnetID`,.

File: docs/resources/cbh_instance.md (hunk 6 above)
Comment (houpeng80) [reply docs/resources/cbh_instance.md]: fixed

File: docs/resources/cbh_instance.md (hunk 9 above)
Comment (houpeng80) [reply docs/resources/cbh_instance.md]: done

File: docs/resources/cbh_instance.md (hunk 10 above)
Comment (houpeng80) [reply docs/resources/cbh_instance.md]: done

File: huaweicloud/services/cbh/resource_huaweicloud_cbh_instance.go [hunk 12]
```diff
@@ -199,454 +140,186 @@ this field is not empty.`,
...
 	for _, v := range instances {
 		instance := v.(map[string]interface{})
-		if instance["resource_id"].(string) == resourceId {
-			return instance["instance_id"].(string), nil
+		if instance["resourceId"] != nil && instance["resourceId"].(string) == resourceId {
+			return instance["instanceId"].(string), nil
 		}
 	}
-	return "", diag.Errorf("error get instance by resource_id: %s", resourceId)
+	return "", fmt.Errorf("error get instance_id by resource_id: %s", resourceId)
```
Comment (ShiChangkuo) [inline huaweicloud/services/cbh/resource_huaweicloud_cbh_instance.go]:
```suggestion
	return "", fmt.Errorf("error getting instance_id by resource_id: %s", resourceId)
```

File: huaweicloud/services/cbh/resource_huaweicloud_cbh_instance.go [hunk 13]
```diff
@@ -736,210 +429,202 @@ func buildCreateInstanceNicsChildBody(d *schema.ResourceData) interface{} {
...
+	getCbhOrderProductIdResp, err := getCbhOrderProductIdClient.Request("POST",
+		getCbhOrderProductIdPath, &getCbhOrderProductIdOpt)
 
-	raw := rawParams[0].(map[string]interface{})
-	params := map[string]interface{}{
-		"volume_type":  utils.ValueIngoreEmpty(raw["type"]),
-		"size":         utils.ValueIngoreEmpty(raw["size"]),
-		"extend_param": utils.ValueIngoreEmpty(raw["extend_param"]),
+	if err != nil {
+		return "", fmt.Errorf("error get CBH order product id: %s", err)
```
Comment (ShiChangkuo) [inline huaweicloud/services/cbh/resource_huaweicloud_cbh_instance.go]:
```suggestion
		return "", fmt.Errorf("error getting CBH order product id: %s", err)
```

File: huaweicloud/services/cbh/resource_huaweicloud_cbh_instance.go [hunk 14]
```diff
@@ -1051,117 +712,107 @@ func updatePassword(d *schema.ResourceData, config *config.Config, region, produ
...
-				d.Set("alter_permit", instance["alter_permit"]),
-				d.Set("bastion_version", instance["bastion_version"]),
-				d.Set("new_bastion_version", instance["new_bastion_version"]),
-				d.Set("instance_status", instance["instance_status"]),
-				d.Set("description", instance["description"]),
-				d.Set("auto_renew", instance["auto_renew"]),
-			)
-			break
+		if instance["instanceId"].(string) != d.Id() {
+			continue
```
Comment (ShiChangkuo) [inline huaweicloud/services/cbh/resource_huaweicloud_cbh_instance.go:759]: missing CheckDeleted

File: docs/resources/cbh_instance.md [hunk 15]
```diff
@@ -69,203 +68,55 @@ The following arguments are supported:
...
 
-* `create` - Default is 30 minute.
-* `delete` - Default is 10 minute.
+* `create` - Default is 60 minutes.
+* `delete` - Default is 30 minutes.
 
 ## Import
 
-The cbh instance can be imported using the `id`, e.g.
+The CBH instance can be imported using the `id`, e.g.
```
Comment (ShiChangkuo) [inline docs/resources/cbh_instance.md:118]: some parameters can not be imported, please add a note here.

**Comment by houpeng80** [comment]
make testacc TEST='./huaweicloud/services/acceptance/cbh/' TESTARGS='-run TestAccCBHInstance_basic'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/services/acceptance/cbh/ -v -run TestAccCBHInstance_basic -timeout 360m -parallel 4
=== RUN   TestAccCBHInstance_basic
=== PAUSE TestAccCBHInstance_basic
=== CONT  TestAccCBHInstance_basic
--- PASS: TestAccCBHInstance_basic (1092.33s) 
PASS                                                                                                              
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/cbh       1092.383s

File: docs/resources/cbh_instance.md (hunk 15 above)
Comment (houpeng80) [reply docs/resources/cbh_instance.md:118]: done

File: huaweicloud/services/cbh/resource_huaweicloud_cbh_instance.go (hunk 14 above)
Comment (houpeng80) [reply huaweicloud/services/cbh/resource_huaweicloud_cbh_instance.go:759]: done

File: huaweicloud/services/cbh/resource_huaweicloud_cbh_instance.go (hunk 13 above)
Comment (houpeng80) [reply huaweicloud/services/cbh/resource_huaweicloud_cbh_instance.go]: fixed

File: huaweicloud/services/cbh/resource_huaweicloud_cbh_instance.go (hunk 12 above)
Comment (houpeng80) [reply huaweicloud/services/cbh/resource_huaweicloud_cbh_instance.go]: fixed

**Pull request merged**